### PR TITLE
disable Manage channels for user without permission

### PR DESCRIPTION
**Pull request by AlicjaSzu**
I want to merge this change because...

- it hides MANAGE channels button when user doesn't have proper permission.

### Screenshots
Screenshot 2020-11-24 at 16 26 24
Screenshot 2020-11-24 at 16 26 34

### Pull Request Checklist

1. [x] All visible strings are translated with proper context.
1. [x] All data-formatting is locale-aware (dates, numbers, and so on).
1. [x] Translated strings are extracted.
1. [x] Number of API calls is optimized.
1. [x] The changes are tested.
1. [x] Data-test are added for new elements.
1. [x] Type definitions are up to date.
1. [ ] Changes are mentioned in the changelog.

### Test environment config

API_URI=https://master.staging.saleor.rocks/graphql/

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-1638 Product availability section management not disabled for user without permissions](https://app.clickup.com/t/2549495/SALEOR-1638)

File: src/components/ChannelsAvailability/ChannelsAvailabilityWithPermission.tsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import { hasPermission } from "@saleor/auth/misc";
```
Comment (mmarkusik) [inline src/components/ChannelsAvailability/ChannelsAvailabilityWithPermission.tsx:1]: @AlicjaSzu an meself figured it'd make more sense to make a general permissions wrapper component for these kind of situations

**Review by mmarkusik — APPROVED** [review]

File: src/components/ChannelsAvailability/ChannelsAvailabilityWithPermission.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  "hasManageChannelsPermission"
+>> = props => {
+  const { user } = useUser();
+  const hasManageChannelsPermission =
+    !!user && hasPermission(PermissionEnum.MANAGE_CHANNELS, user);
+
+  return (
+    <ChannelsAvailability
+      {...props}
+      hasManageChannelsPermission={hasManageChannelsPermission}
```
Comment (orzechdev) [inline src/components/ChannelsAvailability/ChannelsAvailabilityWithPermission.tsx:20]: `ChannelsAvailability` doesn't need to know anything about permissions, I would change it to something like: 
```suggestion
      manageChannels={hasManageChannelsPermission}
```
as the `ChannelsAvailability` component would only care about whether managing channels UI within it should be displayed, while a decision on when it should be displayed, e.g. when permission is available, would be left for upper layer components.

**Review by orzechdev — CHANGES_REQUESTED** [review]

**Review by dominik-zeglen — CHANGES_REQUESTED** [review]
Replicating `src/components/RequirePermissions.tsx`

**Review by krzysztofwolski — APPROVED** [review]

**Review by orzechdev — APPROVED** [review]

**Pull request merged**